data/summary/summary.go

```diff
@@ -55,21 +58,31 @@ var _ Summarizer[types.CGMStats, *types.CGMStats] = &GlucoseSummarizer[types.CGM
 var _ Summarizer[types.BGMStats, *types.BGMStats] = &GlucoseSummarizer[types.BGMStats, *types.BGMStats]{}
 
 type GlucoseSummarizer[T types.Stats, A types.StatsPt[T]] struct {
-	deviceData dataStore.DataRepository
-	summaries  *store.Repo[T, A]
+	userData  types.DeviceDataManager
+	summaries *store.Repo[T, A]
+}
+
+type Manager struct {
+	summaries *store.TypelessRepo
 }
 
-func NewBGMSummarizer(collection *storeStructuredMongo.Repository, deviceData dataStore.DataRepository) Summarizer[types.BGMStats, *types.BGMStats] {
+func NewBGMSummarizer(collection *storeStructuredMongo.Repository, dataRepo dataStore.DataRepository) Summarizer[types.BGMStats, *types.BGMStats] {
 	return &GlucoseSummarizer[types.BGMStats, *types.BGMStats]{
-		deviceData: deviceData,
-		summaries:  store.New[types.BGMStats, *types.BGMStats](collection),
+		userData:  dataRepo,
+		summaries: store.New[types.BGMStats, *types.BGMStats](collection),
 	}
 }
 
-func NewCGMSummarizer(collection *storeStructuredMongo.Repository, deviceData dataStore.DataRepository) Summarizer[types.CGMStats, *types.CGMStats] {
+func NewCGMSummarizer(collection *storeStructuredMongo.Repository, dataRepo dataStore.DataRepository) Summarizer[types.CGMStats, *types.CGMStats] {
 	return &GlucoseSummarizer[types.CGMStats, *types.CGMStats]{
-		deviceData: deviceData,
-		summaries:  store.New[types.CGMStats, *types.CGMStats](collection),
+		userData:  dataRepo,
+		summaries: store.New[types.CGMStats, *types.CGMStats](collection),
+	}
+}
+
+func NewManager(collection *storeStructuredMongo.Repository) *Manager {
```

Comment:
"Manager" is very generic. Consider moving all of the business logic from the API layer to this struct and renaming this to `ClinicRealtimeDataReportGenerator`.

Reply:
`Reporter` class functionality now replaces this